### PR TITLE
My/remove unwrap close #28

### LINKED ISSUE #28: remove unwraps in worker code.
The worker crashes if if `unwrap`s on an error or a none. There have been some unwraps introduced in the oli code, which need to be removed.

### PR DESCRIPTION
**Pull request by m-yahya**

### CODE REVIEW

File: app-libs/stf/src/best_energy_helpers.rs [hunk 1]
```diff
@@ -61,13 +73,27 @@ mod test {
...
+					None => {
+						//
+					},
+				}
+			},
+			Err(e) => Err(StfError::Dispatch(format!(
+				"Getting Merkle Proof {:?}. Error: {:?}",
+				orders, e
+			))),
+		}
```
Comment (clangenb) [inline app-libs/stf/src/best_energy_helpers.rs:96]: In the tests you leave the unwraps. :)

File: app-libs/stf/src/best_energy_helpers.rs [hunk 2]
```diff
@@ -100,7 +126,13 @@ pub fn default_orders() -> Vec<Order> {
...
     }]"#;
 
-	serde_json::from_str(orders_raw).unwrap()
+	match serde_json::from_str::<Vec<Order>>(orders_raw) {
+		Ok(orders) => orders,
+		Err(e) => {
+			info!("Creating raw orders {:?}. Error: {:?}", orders_raw, e);
+			Vec::new()
+		},
+	}
```
Comment (clangenb) [inline app-libs/stf/src/best_energy_helpers.rs:135]: Here too, this is only used for tests.

File: app-libs/stf/src/best_energy_helpers.rs [hunk 3]
```diff
@@ -11,20 +12,32 @@ pub static RESULTS_DIR: &str = "./records/market_results";
...
+	match serde_json::to_string(&orders) {
+		Ok(serialized_orders) => match fs::write(&orders_path, serialized_orders) {
+			Ok(_) => Ok(()),
+			Err(e) =>
+				Err(StfError::Dispatch(format!("Writing orders {:?}. Error: {:?}", orders, e))),
+		},
+		Err(e) =>
+			Err(StfError::Dispatch(format!("Serializing orders {:?}. Error: {:?}", orders, e))),
+	}
 }
```
Comment (clangenb) [inline app-libs/stf/src/best_energy_helpers.rs:24]: Double nesting is hard to read. It is better pracice to keep this sequential:

```rust
	let orders_serialized = serde_json::to_string(&orders)
		.map_err(|e| StfError::Dispatch(format!("Serializing orders {:?}. Error: {:?}", orders, e)))?;

	fs::write(&orders_path, orders_serialized)
		.map_err(|e| StfError::Dispatch(format!("Writing orders {:?}. Error: {:?}", orders, e)))
```

File: app-libs/stf/src/getter.rs [hunk 4]
```diff
@@ -214,7 +214,15 @@ impl ExecuteGetter for Getter {
...
 					let orders_encoded: Vec<Vec<u8>> = orders.iter().map(|o| o.encode()).collect();
 
-					let leaf_index_u32: u32 = (*leaf_index).into();
+					let leaf_index: usize = match (*leaf_index).try_into() {
+						Ok(index) => index,
+						Err(_) => {
+							info!("Error converting Leaf Index to usize: {})", leaf_index);
+							return None
+						},
+					};
```
Comment (clangenb) [inline app-libs/stf/src/getter.rs:223]: 👍

**Review by clangenb — COMMENTED** [review]
Some style comments, but it looks good otherwise. :)

File: app-libs/stf/src/best_energy_helpers.rs (hunk 1 above)
Comment (m-yahya) [reply app-libs/stf/src/best_energy_helpers.rs:96]: done

File: app-libs/stf/src/best_energy_helpers.rs (hunk 2 above)
Comment (m-yahya) [reply app-libs/stf/src/best_energy_helpers.rs:135]: done

File: app-libs/stf/src/trusted_call.rs [hunk 5]
```diff
@@ -283,7 +281,14 @@ where
...
 
-				create_dir_all(ORDERS_DIR).unwrap();
+				match fs::create_dir_all(ORDERS_DIR) {
+					Ok(_) => {},
+					Err(e) =>
+						return Err(StfError::Dispatch(format!(
+							"Creating orders directory. Error: {:?}",
+							e
+						))),
+				}
```
[the remark this answers is not in the document]
Comment (m-yahya) [reply app-libs/stf/src/trusted_call.rs]: done

File: app-libs/stf/src/best_energy_helpers.rs [hunk 6]
```diff
@@ -11,20 +12,32 @@ pub static RESULTS_DIR: &str = "./records/market_results";
...
 				market_results, e
-			))
+			))),
 		},
-	)
+		Err(e) => Err(StfError::Dispatch(format!(
+			"Serializing market results {:?}. Error: {:?}",
+			market_results, e
+		))),
+	}
```
[the remark this answers is not in the document]
Comment (m-yahya) [reply app-libs/stf/src/best_energy_helpers.rs]: done

File: app-libs/stf/src/best_energy_helpers.rs (hunk 3 above)
Comment (m-yahya) [reply app-libs/stf/src/best_energy_helpers.rs]: done

File: app-libs/stf/src/getter.rs [hunk 7]
```diff
@@ -214,7 +214,15 @@ impl ExecuteGetter for Getter {
...
+
+					let leaf_index_u32 = leaf_index as u32;
 					if leaf_index_u32 >= orders.len() as u32 {
```
Comment (clangenb) [inline app-libs/stf/src/getter.rs:226]: Here, you can actually just use the original leaf index if you rename the variable above and then you don't need to cast `orders.len()`

```rust
if leaf_index >= orders.len()
```

**Review by clangenb — COMMENTED** [review]
One minor detail, but it looks good otherwise!

File: app-libs/stf/src/getter.rs (hunk 7 above)
Comment (m-yahya) [reply app-libs/stf/src/getter.rs]: done

**Review by clangenb — APPROVED** [review]
Cool, looks good to me now!

**Pull request merged**